src/GraphQL/Utilities/Visitors/SchemaValidationVisitor.cs

```diff
@@ -233,6 +238,28 @@ public override void VisitDirectiveArgumentDefinition(QueryArgument argument, Di
             // 4.2
             if (!argument.ResolvedType.IsInputType())
                 throw new InvalidOperationException($"The argument '{argument.Name}' of directive '{type.Name}' must be an input type.");
+
+            // validate default
+            if (argument.DefaultValue is GraphQLValue value)
+            {
+                argument.DefaultValue = Execution.ExecutionHelper.CoerceValue(argument.ResolvedType, Language.CoreToVanillaConverter.Value(value)).Value;
+            }
+            else if (argument.DefaultValue != null && !argument.ResolvedType.IsValidDefault(argument.DefaultValue))
+            {
+                throw new InvalidOperationException($"The default value of argument '{argument.Name}' of directive '{type.Name}' is invalid.");
+            }
+        }
+
+        private void ValidateQueryArgumentDefaultValue(QueryArgument argument, FieldType field, INamedType type)
+        {
+            if (argument.DefaultValue is GraphQLValue value)
+            {
+                argument.DefaultValue = Execution.ExecutionHelper.CoerceValue(argument.ResolvedType, Language.CoreToVanillaConverter.Value(value)).Value;
```

Comment:
Please add note in class summary about possible modification of default values for arguments and input object fields.

Reply:
Done; please review

Reply:
Good.